code/game/machinery/shuttle/shuttle_heater.dm

```diff
@@ -0,0 +1,136 @@
+//-----------------------------------------------
+//--------------Engine Heaters-------------------
+//This uses atmospherics, much like a thermomachine,
+//but instead of changing temp, it stores plasma and uses
+//it for the engine
+//-----------------------------------------------
+/obj/machinery/atmospherics/components/unary/shuttle
+	name = "shuttle atmospherics device"
+	desc = "This does something to do with shuttle atmospherics"
+	icon_state = "heater"
+	icon = 'icons/turf/shuttle.dmi'
+
+/obj/machinery/atmospherics/components/unary/shuttle/proc/check_setup(var/affectSurrounding = TRUE)
+	if(!affectSurrounding)
+		return
+	//Don't update if not on shuttle, to prevent lagging out the server in space
+	if(!istype(get_turf(src), /area/shuttle/custom))
```

Comment:
`get_area(src)`?

Reply:
Oh missed this, was to prevent people building them in space and lagging server, so less important now but fixing anyway. I think its like this in engine too